vllm/config/parallel.py

```diff
@@ -223,6 +223,66 @@ class is dynamically inherited by the worker class. This is used to inject
         should only be set by API server scale-out.
     """
 
+    @field_validator("data_parallel_backend")
+    @classmethod
+    def _validate_data_parallel_backend(cls, data_parallel_backend: str) -> str:
+        if data_parallel_backend not in {"mp", "ray"}:
+            raise ValueError(
+                "data_parallel_backend must be either 'mp' or 'ray', "
+                f"but got {data_parallel_backend}."
+            )
+        return data_parallel_backend
+
+    @model_validator(mode="after")
+    def _configure_parallel(self) -> Self:
```

Comment:
**Note:** I only moved the logic concerned with pure validation - not logic for overriding fields. Let me know if you think everything should be moved into the model validator, or if we should split this validator into several smaller and logically separate `model_validators`

Reply:
This seems like a sensible distinction to make. The `model_validator` checks that the values passed are valid, the `__post_init__` does extra initialisation.

One nit about the name for consistency:

```suggestion
    def _validate_parallel_config(self) -> Self:
```